elastic/logstash
# Add API SSL supported protocols setting

## Release notes


Added a new Logstash API setting to configure the SSL/TLS supported protocols.

## What does this PR do?



This PR adds a new Logstash API setting to configure the SSL/TLS supported protocols (`api.ssl.supported_protocols`).

By default, this setting has no default value so it will continue using the puma defaults [`TLSv1`, `TLSv1.1`, `TLSv1.2`] + the new recently added `TLSv1.3`.

In order to validate the possible values for this setting, a new type [StringArray](https://github.com/elastic/logstash/pull/15166/files#diff-291759c068472d4dd44be74d9d757c239e6e457b105df0f5d58218cc6a2b507f) was created.

## Why is it important/What is the impact to the user?



We've recently added support to `TLSv1.3` on Logstash's API endpoint (https://github.com/elastic/logstash/pull/15158). This PR will allow users to configure/fine-tune its security by configuring the supported protocols.

## Checklist




## How to test this PR locally


- Run Logstash with SSL enabled (`api.ssl.enabled: true`) + keystore configured (`api.ssl.keystore.path`, `api.ssl.keystore.password`)
- Configure the supported protocols to only accept `TLSv1.3` (`api.ssl.supported_protocols: TLSv1.3`)
- Connecting using TLS 1.2 should fail:
```shell
openssl s_client -connect localhost:9600 -tls1_2
```
- Connecting using TLS 1.3 should succeed:
```shell
openssl s_client -connect localhost:9600 -tls1_3
```

## Related issues


- Closes https://github.com/elastic/logstash/issues/15156
